src/vs/server/node/server.main.ts

```diff
@@ -571,10 +562,18 @@ export async function main(options: IServerOptions): Promise<void> {
 				const parsedUrl = url.parse(req.url, true);
 				const pathname = parsedUrl.pathname;
 
-				if (options.handleRequest && await instantiationService.invokeFunction(accessor => options.handleRequest!(pathname, req, res, accessor, channelServer))) {
+				let verified = true;
+				if (options.verifyRequest) {
+					verified = await instantiationService.invokeFunction(accessor => options.verifyRequest!(req, accessor));
+				}
+				const verifiedReq = req as VerifiedIncomingMessage;
+				verifiedReq.verified = verified;
+				if (options.handleRequest && await instantiationService.invokeFunction(accessor => options.handleRequest!(pathname, verifiedReq, res, accessor, channelServer))) {
 					return;
 				}
-
+				if (!verifiedReq.verified) {
```

Comment:
Can it be just:
```ts
if (options.verifyRequest && !(await instantiationService.invokeFunction(accessor => options.verifyRequest(req, accessor))) {
    return serveError(req, res, 401, 'Unauthorized');
}
```

Reply:
Well, not really (I would have liked to implement it that way). We still need to send the login page back, if we haven't verified the client yet. Also, we need to change the flag, if we verify during the request (like receiving a form data request with the password)

Reply:
Can we hide then `serveError(req, res, 401, 'Unauthorized')` behind `verifyRequest`? i.e.:
```ts
if (options.verifyRequest && !(await instantiationService.invokeFunction(accessor => options.verifyRequest(req, accessor))) {
    return;
}
```

Reply:
Alright, done 👍